t/t3438-rebase-git-logs-symlink.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+
+test_description='git rebase when .git/logs is a symlink
+
+This test runs a "git rebase" in a repository where .git/logs
+is a symlink and checks that the symlink is not removed.
+If the system does not support symlinks, the whole test will
+be skipped. On Windows, the symlinks may be enabled by granting
+the "Create Symbolic Links" user right and settting the environment
+variable MSYS to "winsymlinks:nativestrict" in Git Bash.'
+
+. ./test-lib.sh
+
+if ! test_have_prereq SYMLINKS
+	then
+	    skip_all='Symlinks support is required for this test'
+	    test_done
+	fi
+
+# Create a rebase situation with 2 commits on a branch and another commit which
+# diverges from them. The 2 commits are created on a branch called
+# "rebase-reference" so that the rebase command that will be used later does
+# not depend on the name of the default branch (master, main, etc.).
+test_expect_success 'Create commits to prepare rebase' '
+    git switch --create rebase-reference &&
+    git commit --allow-empty -m "Initial commit" &&
+    git commit --allow-empty -m "Second commit" &&
+    git switch --create branch-to-rebase HEAD~1 &&
+    git commit --allow-empty -m "Dummy commit to rebase"
```

Comment:
That's a lot of commands to essentially set up the stage for a simple rebase. How about this instead?

```sh
test_commit initial &&
test_commit second
```

The test case can then consist of the command `git rebase -f HEAD^` followed by a test that verifies that `.git/logs` is still a symbolic link.

Reply:
`git rebase -f HEAD^` is much better, thanks for the suggestion :)